internal/platform/select_darwin.go

```diff
@@ -0,0 +1,56 @@
+//go:build darwin || unix
+
+package platform
+
+import (
+	"syscall"
+	"time"
+)
+
+func SelectTimeout(timeout time.Duration) error {
+	return Select(0, nil, nil, nil, timeout)
+}
+
+func SelectStdin(timeout time.Duration) (bool, error) {
+	fdSet := &FDSet{}
+	fdSet.Set(0)
```

Comment:
please use a constant for stdin's fd. When this is done at the call site for wasi, it will be easier as we have FdStdin.